src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertCollectionLoader.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class CertCollectionLoader : ILoaderPal
+    {
+        private List<ICertificatePal> _certs;
+
+        internal CertCollectionLoader(List<ICertificatePal> certs)
+        {
+            _certs = certs;
+        }
+
+        public void Dispose()
+        {
+            // If there're still certificates, dispose them.
+            _certs?.DisposeAll();
+        }
+
+        public void LoadTo(X509Certificate2Collection collection)
```

Comment:
Maybe "MoveTo" since this seems to have move semantics rather than copy semantics?